### DISCUSSION TITLE
keycloak 21 custom image building fails

### DISCUSSION THREAD

**Discussion by kkcmadhu** [keycloak/keycloak#17435] [Q&A]
i use github pipeline for build ,
when build my custom keycloak image n top of keycloak 17 every thing goes fine.
when i attempt to do the same on keycloak 21, my image building task/job fails wit insufficient memory

I have few spis and a login theme. which i attempt to copy to keycloak  providers and themes directory,nothing more..

Any ides what could be going wrong here?

`
Step 1/10 : FROM quay.io/keycloak/keycloak:21.0.1 as builder
21.0.1: Pulling from keycloak/keycloak
50a4297a5857: Pulling fs layer
0baea816215f: Pulling fs layer
8d7e65b21eb5: Pulling fs layer
fc2f56e2fd45: Pulling fs layer
fc2f56e2fd45: Waiting
50a4297a5857: Verifying Checksum
50a4297a5857: Download complete
fc2f56e2fd45: Verifying Checksum
fc2f56e2fd45: Download complete
0baea816215f: Verifying Checksum
0baea816215f: Download complete
8d7e65b21eb5: Verifying Checksum
8d7e65b21eb5: Download complete
50a4297a5857: Pull complete
0baea816215f: Pull complete
8d7e65b21eb5: Pull complete
fc2f56e2fd45: Pull complete
Digest: sha256:057e1264cae9adbd9be65235d0c837087b0accc183275803b7da81b1b7a2a94c
Status: Downloaded newer image for quay.io/keycloak/keycloak:21.0.1
 ---> 2cb4b870707e
Step 2/10 : ADD  ./my-custom/spi 	   /opt/keycloak/providers
 ---> 0e21477de3f9
Step 3/10 : ADD  ./my-custom/themes/my-default-theme   /opt/keycloak/themes/my-default-theme
 ---> 50a999601d9b
Step 4/10 : RUN /opt/keycloak/bin/kc.sh build --db=mysql --features="token-exchange,preview,admin-fine-grained-authz" --http-relative-path=/auth --health-enabled=true --metrics-enabled=true --cache ispn  --cache-stack=kubernetes --spi-login-provider=my-loginprovider --spi-login-freemarker-enabled=false --spi-brute-force-protector-provider=my-default-brute-force-detector --spi-brute-force-protector-default-brute-force-detector-enabled=true
 ---> Running in 7bd37ea9cd89
Warning: [0.002s][warning][os,thread] Failed to start thread "GC Thread#0" - pthread_create failed (EPERM) for attributes: stacksize: 1024k, guardsize: 4k, detached.

 There is insufficient memory for the Java Runtime Environment to continue.
 Cannot create worker GC thread. Out of system resources.
 An error report file with more information is saved as:
 /tmp/hs_err_pid1.log
The command '/bin/sh -c /opt/keycloak/bin/kc.sh build --db=mysql --features="token-exchange,preview,admin-fine-grained-authz" --http-relative-path=/auth --health-enabled=true --metrics-enabled=true --cache ispn  --cache-stack=kubernetes --spi-login-provider=my-loginprovider --spi-login-freemarker-enabled=false --spi-brute-force-protector-provider=my-default-brute-force-detector --spi-brute-force-protector-default-brute-force-detector-enabled=true' returned a non-zero code: 1
Error: Process completed with exit code 1.`

**Comment by DAHAG-ArisNourbakhsh** [comment]
might be related to #17404

**Reply by kkcmadhu** [reply]
i think so!
